src/integration-test/java/uk/ac/ebi/tsi/portal/api/team/TeamRestControllerIT.java

```diff
@@ -86,21 +99,112 @@ public void can_get_a_team() throws Exception{
 
 			logger.info("Response "  +response);
 		}
-		
-		
-		public void can_create_a_team() throws Exception{
-			Team team = new Team();
-			team.setName(teamName);
-			String json = mapper.writeValueAsString(team);
+
+	@Test
+	public void team_owner_can_add_team_contact_emails() throws Exception {
+
+		String emails = "contact1@ebi,contact2@ebi";
+		mockMvc.perform(
+				put("/team/" + "test-team1" + "/contactemail/")
+						.headers(createHeaders(getToken(ajayUserName, ajayPassword)))
+						.contentType(MediaType.APPLICATION_JSON)
+						.content(emails)
+						.accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk());
+
+	}
+
+	@Test
+	public void team_owner_can_remove_team_contact_emails() throws Exception {
+
+		String deleteURL = "/team/" + "test-team1" + "/contactemail/" + "contact@ebi";
+		mockMvc.perform(
+				delete(deleteURL)
+						.headers(createHeaders(getToken(ajayUserName, ajayPassword)))
+						.contentType(MediaType.APPLICATION_JSON)
+						.accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isNoContent());
+	}
+
+	@Test
+	public void non_team_owner_cannot_add_team_contact_emails() throws Exception {
+
+		String emails = "contact1@ebi,contact2@ebi";
+		try {
 			mockMvc.perform(
-					post("/team")
-					.headers(createHeaders(token))
-					.contentType(MediaType.APPLICATION_JSON)
-					.content(json)
-					.accept(MediaType.APPLICATION_JSON))
-			.andExpect(jsonPath("name").value(teamName))
-			.andReturn();
+					put("/team/" + "test-team1" + "/contactemail/")
+							.headers(createHeaders(token))
+							.contentType(MediaType.APPLICATION_JSON)
+							.content(emails)
+							.accept(MediaType.APPLICATION_JSON))
+			;
+		} catch (NestedServletException e) {
+			assertEquals(e.getCause().getClass(), TeamAccessDeniedException.class);
```

Comment:
What is going on here? Why this `NestedServletException`? How does such a use case work from a user perspective?

Reply:
{
    "timestamp": 1615461288001,
    "status": 403,
    "error": "Forbidden",
    "exception": "uk.ac.ebi.tsc.portal.api.team.service.TeamAccessDeniedException",
    "message": "Only owner of the team ssxs can access contact emails",
    "path": "/team/ssxs/contactemail/sdsd"
} Nothing changes from user's perspective, the above is what he will see. It's a runtime exception not declared in method signature.

Reply:
OK - it looks like a known topic for mockMVC https://stackoverflow.com/questions/44900646/how-to-prevent-nestedservletexception-when-testing-spring-endpoints (for some reason I do not see the same behaviour in other Spring Boot apps). We might investigate it another time.